crates/ruff_python_formatter/src/other/arguments.rs

```diff
@@ -1,12 +1,188 @@
-use crate::{not_yet_implemented, FormatNodeRule, PyFormatter};
-use ruff_formatter::{write, Buffer, FormatResult};
-use rustpython_parser::ast::Arguments;
+use crate::comments::{dangling_node_comments, leading_node_comments};
+use crate::context::NodeLevel;
+use crate::other::arg::ArgumentKind;
+use crate::prelude::*;
+use crate::trivia::{first_non_trivia_token, SimpleTokenizer, Token, TokenKind};
+use crate::FormatNodeRule;
+use ruff_formatter::{format_args, write, FormatError};
+use ruff_python_ast::node::{AnyNodeRef, AstNode};
+use rustpython_parser::ast::{Arg, Arguments, Expr, Ranged};
+use std::usize;
 
 #[derive(Default)]
 pub struct FormatArguments;
 
 impl FormatNodeRule<Arguments> for FormatArguments {
     fn fmt_fields(&self, item: &Arguments, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let Arguments {
+            range: _,
+            posonlyargs,
+            args,
+            defaults,
+            vararg,
+            kwonlyargs,
+            kw_defaults,
+            kwarg,
+        } = item;
+
+        let saved_level = f.context().node_level();
+        f.context_mut().set_node_level(NodeLevel::Expression);
+
+        let format_inner = format_with(|f: &mut PyFormatter| {
+            let separator = format_with(|f| write!(f, [text(","), soft_line_break_or_space()]));
+            let mut joiner = f.join_with(separator);
+            let mut last_node: Option<AnyNodeRef> = None;
+
+            let mut defaults = std::iter::repeat(None)
+                .take(posonlyargs.len() + args.len() - defaults.len())
+                .chain(defaults.iter().map(Some));
+
+            for positional in posonlyargs {
+                let default = defaults.next().ok_or(FormatError::SyntaxError)?;
+                joiner.entry(&ArgumentWithDefault {
+                    argument: positional,
+                    default,
+                });
+
+                last_node = Some(default.map_or_else(|| positional.into(), AnyNodeRef::from));
+            }
+
+            if !posonlyargs.is_empty() {
+                joiner.entry(&text("/"));
```

Comment:
so confusing that pos only arguments are in front of the slash, but keyword arguments are after the star (your code is correct)